public/pages/utils/helpers.ts

```diff
@@ -13,52 +13,60 @@
  * permissions and limitations under the License.
  */
 
-import { CatIndex, IndexAlias } from '../../../../../../server/models/types';
+import { CatIndex, IndexAlias } from '../../../server/models/types';
 
 export function sanitizeSearchText(searchValue: string): string {
   if (!searchValue || searchValue == '*') {
     return '';
   }
   if (canAppendWildcard(searchValue)) {
-    return `${searchValue}*`;
+    return `*${searchValue}*`;
   } else {
     return searchValue;
   }
 }
-export function canAppendWildcard(searchValue: string): boolean {
-  // If it's not a letter, number or is something longer, reject it
-  if (
-    !searchValue ||
-    !/[a-z0-9]/i.test(searchValue) ||
-    searchValue.length !== 1
-  ) {
+function canAppendWildcard(searchValue: string): boolean {
+  // If it's not a letter or number, reject it
+  if (!searchValue || !/[a-z0-9]/i.test(searchValue)) {
     return false;
   }
   return true;
 }
 
-export const isSystemIndices = (index: string) => {
+const isSystemIndices = (index: string) => {
   if (!index) {
     return index;
   }
   return !index.startsWith('.');
 };
 
-export function getVisibleOptions(indices: CatIndex[], aliases: IndexAlias[]) {
+export function getVisibleOptions(indices: CatIndex[], aliases: IndexAlias[], maxLen: number) {
   const visibleIndices = indices
     .filter(value => isSystemIndices(value.index))
     .map(value => ({ label: value.index, health: value.health }));
   const visibleAliases = aliases
     .filter(value => isSystemIndices(value.alias))
     .map(value => ({ label: value.alias }));
+
+  // only returning the top maxLen results to display
```

Comment:
Could you explain with this example what you're trying to achieve ?

Reply:
Me and Yizhe were discussing the possibility of a customer having a large number of indices, so we wanted to limit the amount shown in the combo box so it wouldn't extend off the screen. I've set a default value of 20 for now. I've manually tested this to confirm it works properly.

Reply:
I see now that a scroll bar is introduced once the displayed list is more than 6 or 7. I think that limiting the number of visible indices is probably unnecessary here now. Would you agree that I should remove this limit?